File: MPI_Code/Fortran/scatterv2.f90
PROGRAM scatterv2
USE mpi_f08
IMPLICIT NONE
INTEGER::isend(6)=0, isndcnt(0:2), idisp(0:2)
INTEGER::rcvcnt,myrank
isndcnt=(/1,2,3/)
idisp=(/0,1,3/)
CALL MPI_Init
CALL MPI_Comm_rank(MPI_COMM_WORLD,myrank)
rcvcnt=myrank+1
IF(myrank==0) isend=(/1,2,2,3,3,3/)
PRINT*,'rank:',myrank,'isend=',isend
IF(myrank==0)THEN
  CALL MPI_Scatterv(isend,isndcnt,idisp,MPI_INTEGER,MPI_IN_PLACE,rcvcnt,MPI_INTEGER,0,MPI_COMM_WORLD)
ELSE
  CALL MPI_Scatterv(isend,isndcnt,idisp,MPI_INTEGER,isend(idisp(myrank)+1),rcvcnt,MPI_INTEGER,0,MPI_COMM_WORLD)
ENDIF
PRINT*,'RANK:',myrank,'ISEND=',isend
CALL MPI_Finalize
END PROGRAM scatterv2

File: example_code/mpi/MPI_Code/C/scatterv2.c
#include <stdio.h>
#include <mpi.h>
int main()
{
  int isend[6]={0,}, isndcnt[3]={1,2,3}, idisp[3]={0,1,3};
  int rcvcnt,myrank,i;
  MPI_Init(NULL,NULL);
  MPI_Comm_rank(MPI_COMM_WORLD,&myrank);
  rcvcnt=myrank+1;
  if(myrank==0){
    isend[0]=1,  isend[1]=2,  isend[2]=2;
    isend[3]=3,  isend[4]=3,  isend[5]=3;
  }
  printf("rank: %d  isend=",myrank);
  for(i=0;i<6;i++) printf("  %d",isend[i]);
  printf("\n");
  if(myrank==0){
    MPI_Scatterv(isend,isndcnt,idisp,MPI_INT,MPI_IN_PLACE,rcvcnt,MPI_INT,0,MPI_COMM_WORLD);
  }
  else{
    MPI_Scatterv(isend,isndcnt,idisp,MPI_INT,&isend[idisp[myrank]],rcvcnt,MPI_INT,0,MPI_COMM_WORLD);
  }
  printf("RANK: %d  ISEND=",myrank);
  for(i=0;i<6;i++) printf("  %d",isend[i]);
  printf("\n");

  MPI_Finalize();
  return 0;
}
